Compilation error in latest main (conflicting PRs for oid4vc and changes for EnvironmentDependentFactory)

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

There is compilation failure caused by 2 conflicting PRs (both were green, but compilation failure happened when they are both merged).

For the reference, the PRs were:
oid4vci related changes: https://github.com/keycloak/keycloak/commit/b08c6446018b2c55cb23c1611fa0baf37d64bfaa
Updates in EnvironmentDependentFactory: https://github.com/keycloak/keycloak/commit/3e0a1850708b9f1a07e3292c79ef8c84744f2fc1

The failure is:

```
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/mposolda/IdeaProjects/keycloak/services/src/main/java/org/keycloak/protocol/oid4vc/OID4VCEnvironmentProviderFactory.java:[34,5] method does not override or implement a method from a supertype
[ERROR] /home/mposolda/IdeaProjects/keycloak/services/src/main/java/org/keycloak/protocol/oidc/grants/PreAuthorizedCodeGrantType.java:[105,5] method does not override or implement a method from a supertype
[INFO] 2 errors 
[INFO] -------------------------------------------------------------

```

### Version

latest main

### Regression

- [ ] The issue is a regression

### Expected behavior

Keycloak compiles

### Actual behavior

Compilation error

### How to Reproduce?

Run this on latest Keycloak main:
```
mvn clean install -DskipTests=true
```

### Anything else?

_No response_

--- a/services/src/main/java/org/keycloak/protocol/oid4vc/OID4VCEnvironmentProviderFactory.java
+++ b/services/src/main/java/org/keycloak/protocol/oid4vc/OID4VCEnvironmentProviderFactory.java
@@ -31,8 +31,4 @@ public interface OID4VCEnvironmentProviderFactory extends EnvironmentDependentPr
         return Profile.isFeatureEnabled(Profile.Feature.OID4VC_VCI);
     }
 
-    @Override
-    default boolean isSupported() {
-        return Profile.isFeatureEnabled(Profile.Feature.OID4VC_VCI);
-    }
 }

--- a/services/src/main/java/org/keycloak/protocol/oidc/grants/PreAuthorizedCodeGrantType.java
+++ b/services/src/main/java/org/keycloak/protocol/oidc/grants/PreAuthorizedCodeGrantType.java
@@ -103,11 +103,6 @@ public class PreAuthorizedCodeGrantType extends OAuth2GrantTypeBase implements E
     }
 
     @Override
-    public boolean isSupported() {
-        return Profile.isFeatureEnabled(Profile.Feature.OID4VC_VCI);
-    }
-
-    @Override
     public EventType getEventType() {
         return EventType.CODE_TO_TOKEN;
     }

